util/pkg/vfs/azureblob.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	azureBlobSchema = "azureblob"
+)
+
+// AzureBlobPath is a path in the VFS space backed by Azure Blob.
+type AzureBlobPath struct {
+	client    *azureClient
+	container string
+	key       string
+}
+
+var _ Path = &AzureBlobPath{}
+var _ HasHash = &S3Path{}
```

Comment:
I guess it won't be too different from the S3 one, but still best to keep consistent.
```suggestion
var _ HasHash = &AzureBlobPath{}
```